portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java

```diff
@@ -957,6 +957,7 @@ public void addRecentSite(String userId, String siteId) {
 		}
 
 		recentSiteRepository.deleteByUserIdAndSiteId(userId, siteId);
+		recentSiteRepository.flush();
```

Comment:
@jesusmmp this shouldn't be needed if you find that it is it's usually because there is another issue